### PR TITLE
Feature: Ask answer block

### PR DESCRIPTION
**Pull request by fredalai**
## Description
  Implement Ask UI - Answer Block

### Task
 - [x]  For each step results, the Preview data and show SQL are hideable, it is triggered by the `Collapse` text icon button
 - [x]  Steps list block
 - [x]  ~`<CodeBlock />` support show line numbers & CSS issue~ 
(updated in the https://github.com/Canner/vulcan-sql/pull/360)
 - [x]  Preview Data Table
     - [x]  Show column type icon, column name, and column description
 - [x]  Skeleton Loading
 - [x]  `Save as view`
        `Save as view`  at the bottom of this thread, click and a popup window appears to create a view using the full SQL in Modeling.    
    - Users needs to enter a view display name
    - Show full SQL statement
    - Click `Save` button and a view will be created in Modeling UI
    - Users can click `X` to close the window and cancel to save the view

### UI
截圖 2024-03-07 下午4 35 08

截圖 2024-03-07 下午4 35 37

截圖 2024-03-07 下午4 35 43

 
截圖 2024-03-07 下午4 36 03

截圖 2024-03-07 下午4 36 09

## Issue ticket number

closes #xx

## Additional Context

 - Not connected to the real API yet

### CODE REVIEW

File: packages/admin-ui/styles/components/button.less [hunk 1]
```diff
@@ -0,0 +1,6 @@
+
+.adm-btn {
+  &-gray {
+    .button-variant-primary(@gray-8; @gray-5; @gray-6; @gray-7);
```
Comment (andreashimin) [inline packages/admin-ui/styles/components/button.less:4]: As discussion before, we adjust the button colors to:
- bg: gray-6
- bg-hover: gray-5
- bg-active: gray-7
- color: gray-9

File: packages/admin-ui/components/ask/PreviewData.tsx [hunk 2]
```diff
@@ -0,0 +1,1449 @@
...
+        title: (
+          <>
+            {columnTypeIcon}
+            <Text ellipsis title={column.name} className="ml-1">
+              {column.name}
+            </Text>
+            <br />
+            <Text type="secondary" ellipsis title={column.description}>
+              {column.description}
+            </Text>
```
[the remark this answers is not in the document]
Comment (fredalai) [reply packages/admin-ui/components/ask/PreviewData.tsx:33]: I will remove the ellipsis props of the first <Text /> component for the render column name to follow the column ellipsis and color.

I keep the second <Text /> component with ellipsis props for the render column description to ensure the ellipsis color same as the text color.

File: packages/admin-ui/components/ask/PreviewData.tsx [hunk 3]
```diff
@@ -0,0 +1,1449 @@
+import { useMemo } from 'react';
+import { Typography } from 'antd';
+import { getColumnTypeIcon } from '@vulcan-sql/admin-ui/utils/columnType';
+import PreviewDataContent from '@vulcan-sql/admin-ui/components/pages/modeling/form/PreviewDataContent';
+
+const { Text } = Typography;
+
+const getValueByValueType = (value: any) =>
+  ['boolean', 'object'].includes(typeof value) ? JSON.stringify(value) : value;
```
[the remark this answers is not in the document]
Comment (fredalai) [reply packages/admin-ui/components/ask/PreviewData.tsx:9]: agree. I assumed that the data structure is an Array<>, so I decided to move the getResult func into the component as well and rename it to convertResultData.

**Review by andreashimin — APPROVED** [review]
LGTM

**Pull request merged**